From 8a69831bfe4deb56401ca8b528dd628317315cad Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Tue, 16 Nov 2021 17:35:32 +0800
Subject: [PATCH] soc: ast10x0: Fix the pin define order.

The order of pin_def_list should be the same as GPIO.
e.g.: N4 -> GPIOA0, P4 -> GPIOA1 and so on.
The GPIO driver will use gpio index to request pin.

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: Iea91863cf99370902b17b478eaf658d777d21943
---
 soc/arm/aspeed/ast10x0/pin_def_list.h | 248 +++++++++++---------------
 1 file changed, 108 insertions(+), 140 deletions(-)

diff --git a/soc/arm/aspeed/ast10x0/pin_def_list.h b/soc/arm/aspeed/ast10x0/pin_def_list.h
index 0a7c14be30..71fcd50d4f 100644
--- a/soc/arm/aspeed/ast10x0/pin_def_list.h
+++ b/soc/arm/aspeed/ast10x0/pin_def_list.h
@@ -1,205 +1,173 @@
-PIN_DEFINE(K1)
-PIN_DEFINE(L1)
-PIN_DEFINE(M1)
-PIN_DEFINE(N2)
-PIN_DEFINE(P2)
-PIN_DEFINE(N1)
+/*
+ * The order of this file should be the same as GPIO.
+ * e.g.: N4 -> GPIOA0, P4 -> GPIOA1 and so on.
+ * Please check the ball map of GPIO before you modify this file.
+ */
+PIN_DEFINE(N4)
+PIN_DEFINE(P4)
 PIN_DEFINE(P1)
+PIN_DEFINE(P5)
+PIN_DEFINE(P2)
+PIN_DEFINE(P3)
 PIN_DEFINE(R1)
 PIN_DEFINE(R2)
-PIN_DEFINE(P3)
-PIN_DEFINE(N4)
-PIN_DEFINE(M5)
-PIN_DEFINE(L6)
-PIN_DEFINE(T2)
+PIN_DEFINE(T1)
 PIN_DEFINE(R3)
+PIN_DEFINE(T2)
 PIN_DEFINE(T3)
-PIN_DEFINE(P4)
+PIN_DEFINE(N5)
+PIN_DEFINE(P6)
+PIN_DEFINE(M5)
 PIN_DEFINE(R4)
-PIN_DEFINE(P5)
-PIN_DEFINE(T4)
 PIN_DEFINE(R5)
-PIN_DEFINE(N5)
-PIN_DEFINE(M6)
+PIN_DEFINE(T4)
+PIN_DEFINE(P7)
 PIN_DEFINE(N6)
-PIN_DEFINE(L8)
+PIN_DEFINE(P8)
 PIN_DEFINE(T5)
 PIN_DEFINE(R6)
-PIN_DEFINE(P6)
-PIN_DEFINE(M8)
-PIN_DEFINE(L9)
 PIN_DEFINE(T6)
 PIN_DEFINE(R7)
-PIN_DEFINE(P7)
-PIN_DEFINE(N8)
-PIN_DEFINE(M9)
 PIN_DEFINE(T7)
 PIN_DEFINE(R8)
-PIN_DEFINE(P8)
-PIN_DEFINE(N9)
 PIN_DEFINE(T8)
-PIN_DEFINE(R9)
-PIN_DEFINE(P9)
-PIN_DEFINE(N10)
-PIN_DEFINE(M11)
+PIN_DEFINE(K1)
+PIN_DEFINE(L3)
+PIN_DEFINE(L4)
+PIN_DEFINE(M4)
+PIN_DEFINE(L2)
+PIN_DEFINE(L1)
+PIN_DEFINE(M3)
+PIN_DEFINE(M2)
+PIN_DEFINE(M1)
+PIN_DEFINE(N3)
+PIN_DEFINE(N2)
+PIN_DEFINE(N1)
 PIN_DEFINE(T9)
+PIN_DEFINE(P9)
+PIN_DEFINE(R9)
+PIN_DEFINE(T10)
 PIN_DEFINE(R10)
 PIN_DEFINE(P10)
-PIN_DEFINE(T10)
+PIN_DEFINE(T11)
+PIN_DEFINE(N11)
 PIN_DEFINE(R11)
-PIN_DEFINE(P11)
+PIN_DEFINE(T12)
+PIN_DEFINE(R12)
 PIN_DEFINE(T13)
-PIN_DEFINE(R14)
+PIN_DEFINE(P11)
+PIN_DEFINE(R13)
 PIN_DEFINE(T14)
 PIN_DEFINE(T15)
-PIN_DEFINE(R15)
-PIN_DEFINE(P14)
-PIN_DEFINE(N13)
+PIN_DEFINE(R14)
 PIN_DEFINE(M12)
-PIN_DEFINE(L11)
+PIN_DEFINE(N12)
+PIN_DEFINE(P14)
+PIN_DEFINE(R15)
+PIN_DEFINE(P12)
+PIN_DEFINE(P13)
+PIN_DEFINE(N14)
+PIN_DEFINE(T16)
 PIN_DEFINE(R16)
 PIN_DEFINE(P15)
-PIN_DEFINE(N14)
-PIN_DEFINE(M13)
-PIN_DEFINE(K11)
 PIN_DEFINE(P16)
+PIN_DEFINE(M13)
 PIN_DEFINE(N15)
-PIN_DEFINE(M14)
-PIN_DEFINE(K12)
-PIN_DEFINE(J11)
 PIN_DEFINE(N16)
+PIN_DEFINE(M14)
 PIN_DEFINE(M15)
+PIN_DEFINE(M16)
 PIN_DEFINE(L14)
-PIN_DEFINE(K13)
-PIN_DEFINE(J12)
 PIN_DEFINE(L15)
-PIN_DEFINE(K14)
-PIN_DEFINE(G11)
-PIN_DEFINE(F11)
+PIN_DEFINE(L16)
 PIN_DEFINE(K16)
+PIN_DEFINE(K14)
 PIN_DEFINE(K15)
-PIN_DEFINE(J14)
-PIN_DEFINE(H13)
-PIN_DEFINE(G12)
-PIN_DEFINE(F12)
-PIN_DEFINE(E12)
 PIN_DEFINE(J16)
 PIN_DEFINE(J15)
-PIN_DEFINE(H14)
-PIN_DEFINE(G13)
-PIN_DEFINE(F13)
-PIN_DEFINE(E13)
-PIN_DEFINE(D13)
+PIN_DEFINE(J14)
+PIN_DEFINE(H16)
 PIN_DEFINE(H15)
-PIN_DEFINE(G14)
 PIN_DEFINE(G15)
-PIN_DEFINE(F14)
-PIN_DEFINE(E14)
-PIN_DEFINE(D14)
-PIN_DEFINE(F16)
-PIN_DEFINE(F15)
-PIN_DEFINE(E15)
-PIN_DEFINE(E16)
+PIN_DEFINE(J13)
+PIN_DEFINE(G16)
+PIN_DEFINE(F12)
+PIN_DEFINE(E12)
 PIN_DEFINE(D15)
-PIN_DEFINE(C14)
 PIN_DEFINE(D16)
-PIN_DEFINE(C15)
 PIN_DEFINE(C16)
-PIN_DEFINE(B15)
+PIN_DEFINE(C15)
 PIN_DEFINE(B16)
-PIN_DEFINE(A15)
-PIN_DEFINE(D12)
+PIN_DEFINE(F11)
+PIN_DEFINE(B15)
 PIN_DEFINE(E11)
+PIN_DEFINE(D14)
+PIN_DEFINE(C14)
+PIN_DEFINE(A16)
+PIN_DEFINE(A15)
+PIN_DEFINE(D13)
 PIN_DEFINE(A14)
-PIN_DEFINE(B13)
+PIN_DEFINE(B14)
+PIN_DEFINE(C13)
+PIN_DEFINE(D12)
 PIN_DEFINE(C12)
+PIN_DEFINE(G14)
+PIN_DEFINE(F16)
+PIN_DEFINE(F15)
+PIN_DEFINE(F14)
+PIN_DEFINE(E16)
+PIN_DEFINE(E15)
+PIN_DEFINE(E14)
+PIN_DEFINE(E13)
+PIN_DEFINE(B13)
 PIN_DEFINE(A13)
+PIN_DEFINE(D11)
 PIN_DEFINE(C11)
-PIN_DEFINE(D10)
+PIN_DEFINE(B12)
 PIN_DEFINE(A12)
-PIN_DEFINE(B11)
+PIN_DEFINE(D10)
 PIN_DEFINE(C10)
-PIN_DEFINE(D8)
 PIN_DEFINE(A11)
+PIN_DEFINE(B11)
+PIN_DEFINE(A10)
 PIN_DEFINE(B10)
-PIN_DEFINE(D7)
-PIN_DEFINE(D6)
+PIN_DEFINE(A9)
 PIN_DEFINE(B9)
+PIN_DEFINE(C9)
+PIN_DEFINE(D9)
+PIN_DEFINE(A8)
+PIN_DEFINE(F7)
+PIN_DEFINE(E7)
+PIN_DEFINE(C5)
 PIN_DEFINE(C8)
-PIN_DEFINE(C7)
-PIN_DEFINE(C6)
+PIN_DEFINE(UNUSED0)
+PIN_DEFINE(D8)
 PIN_DEFINE(D5)
-PIN_DEFINE(D4)
-PIN_DEFINE(A9)
-PIN_DEFINE(B8)
-PIN_DEFINE(B7)
-PIN_DEFINE(C5)
-PIN_DEFINE(C4)
-PIN_DEFINE(A8)
-PIN_DEFINE(B6)
-PIN_DEFINE(B5)
-PIN_DEFINE(C3)
 PIN_DEFINE(A7)
+PIN_DEFINE(B6)
+PIN_DEFINE(A5)
 PIN_DEFINE(A6)
+PIN_DEFINE(B5)
 PIN_DEFINE(A4)
+PIN_DEFINE(B4)
+PIN_DEFINE(D4)
+PIN_DEFINE(C3)
 PIN_DEFINE(B3)
 PIN_DEFINE(A3)
-PIN_DEFINE(B2)
 PIN_DEFINE(A2)
-PIN_DEFINE(T12)
-PIN_DEFINE(R13)
+PIN_DEFINE(B2)
 PIN_DEFINE(B1)
+PIN_DEFINE(C1)
+PIN_DEFINE(D3)
 PIN_DEFINE(C2)
-PIN_DEFINE(E3)
 PIN_DEFINE(D1)
-PIN_DEFINE(E2)
-PIN_DEFINE(F3)
-PIN_DEFINE(G4)
-PIN_DEFINE(E1)
-PIN_DEFINE(F2)
+PIN_DEFINE(D2)
+PIN_DEFINE(E3)
+PIN_DEFINE(H2)
+PIN_DEFINE(H1)
 PIN_DEFINE(J4)
-PIN_DEFINE(L5)
-PIN_DEFINE(F1)
-PIN_DEFINE(G2)
-PIN_DEFINE(H3)
-PIN_DEFINE(J3)
+PIN_DEFINE(J2)
+PIN_DEFINE(J1)
+PIN_DEFINE(K3)
 PIN_DEFINE(K4)
-PIN_DEFINE(L4)
-PIN_DEFINE(G1)
-PIN_DEFINE(B4)
-PIN_DEFINE(A10)
-PIN_DEFINE(C9)
-PIN_DEFINE(D9)
-PIN_DEFINE(T1)
-PIN_DEFINE(C13)
-PIN_DEFINE(L3)
-PIN_DEFINE(M4)
-PIN_DEFINE(L2)
-PIN_DEFINE(M3)
-PIN_DEFINE(M2)
-PIN_DEFINE(N3)
-PIN_DEFINE(T11)
-PIN_DEFINE(N11)
-PIN_DEFINE(R12)
-PIN_DEFINE(B12)
-PIN_DEFINE(D11)
-PIN_DEFINE(N12)
-PIN_DEFINE(P12)
-PIN_DEFINE(P13)
-PIN_DEFINE(T16)
-PIN_DEFINE(A16)
-PIN_DEFINE(C1)
-PIN_DEFINE(D3)
-PIN_DEFINE(D2)
 PIN_DEFINE(K2)
-PIN_DEFINE(K3)
-PIN_DEFINE(J1)
-PIN_DEFINE(J2)
-PIN_DEFINE(H1)
-PIN_DEFINE(H2)
-PIN_DEFINE(M16)
-PIN_DEFINE(H16)
-PIN_DEFINE(B14)
-PIN_DEFINE(L16)
-PIN_DEFINE(J13)
-PIN_DEFINE(G16)
-- 
2.25.1

